state-chain/custom-rpc/src/order_fills.rs

```diff
@@ -0,0 +1,179 @@
+use std::collections::HashSet;
+
+use super::*;
+
+use cf_primitives::AccountId;
+use pallet_cf_pools::{AssetPair, OrderId, Pool};
+use state_chain_runtime::Runtime;
+
+#[derive(serde::Serialize, serde::Deserialize, Clone, PartialEq, Eq)]
+pub struct OrderFills {
+	pub(super) fills: Vec<OrderFilled>,
+}
+
+#[derive(serde::Serialize, serde::Deserialize, Clone, PartialEq, Eq)]
+#[serde(rename_all = "snake_case")]
+pub enum OrderFilled {
+	LimitOrder {
+		lp: AccountId,
+		base_asset: Asset,
+		quote_asset: Asset,
+		side: Side,
+		id: U256,
+		tick: Tick,
+		sold: U256,
+		bought: U256,
+		fees: U256,
+		remaining: U256,
+	},
+	RangeOrder {
```

Comment:
it would be great if we can add the amount that was filled by this order

Reply:
I raised an issue for this, unfortunately it doesn't look so simple..